prow/tide/tide.go

```diff
@@ -551,45 +551,46 @@ func accumulateBatch(presubmits map[int]sets.String, prs []PullRequest, pjs []ku
 		if pj.Spec.Type != kube.BatchJob {
 			continue
 		}
-		// If any batch job is pending, return now.
-		if toSimpleState(pj.Status.State) == pendingState {
-			var pending []PullRequest
-			var pendingNums []int
-			for _, pull := range pj.Spec.Refs.Pulls {
-				pending = append(pending, prNums[pull.Number])
-				pendingNums = append(pendingNums, pull.Number)
-			}
-			log.Debugf("no new batch necessary, current batch pending: %v", pendingNums)
-			return nil, pending
-		}
-		// Otherwise, accumulate results.
+		// First validate the batch job's refs.
 		ref := pj.Spec.Refs.String()
-		if _, ok := states[ref]; !ok {
-			states[ref] = &accState{
+		state, ok := states[ref]
```

Comment:
This is a little hard to follow. I think the `if !ok` block is enforcing that we do have `states[ref]` ultimately, we should

```go
if _, ok := states[ref]; !ok {
    state := &accState{
    // ...
    states[ref] = state
}
```

And always then use `states[ref]` in the following lines -- easier to reason about what is in the `state` variable

Reply:
I made this change for consistency with the loop at the end of the function and because I thought it was confusing that we were repeatedly looking up the same value from the map every time we wanted to edit it instead of reusing the value. It is unclear to me why repeating the map lookup everywhere would make things any easier to reason about.

I switched back to repeating the map lookup everywhere though so that we can get this fix PR merged.